### PR TITLE
update linux setup steps

### PR DESCRIPTION
**Pull request by davidsbailey**
I ran into a number of problems when setting up a linux machine for local development from scratch recently. this PR fixes a number of the issues I ran into.

## Testing story

followed the updated instructions for a DeveloperEC2 instance and confirmed that I could get through installing ruby.

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -171,18 +171,30 @@ After setup, read about our [code styleguide](./STYLEGUIDE.md), our [test suites
 Note: Virtual Machine Users should check the [Alternative note](#alternative-use-an-ubuntu-vm) below before starting
 
 1. `sudo apt-get update`
-1. `sudo apt-get install -y git mysql-server mysql-client libmysqlclient-dev libxslt1-dev libssl-dev zlib1g-dev imagemagick libmagickcore-dev libmagickwand-dev openjdk-9-jre-headless libcairo2-dev libjpeg8-dev libpango1.0-dev libgif-dev curl pdftk enscript libsqlite3-dev build-essential redis-server rbenv chromium-browser parallel`
```
Comment (tim-dot-org) [inline SETUP.md:174]: why did you remove pdftk? it looks like its needed for some pegasus tests

**Review by smusoke — APPROVED** [review]
👍

File: SETUP.md (hunk 1 above)
Comment (davidsbailey) [reply SETUP.md:174]: this wasn't available to me on a default Ubuntu 18 instance on EC2, and most developers don't need to run pegasus tests locally very often. however I'm happy to add it back for now, and maybe we can figure out what to do here once we get to the bottom of the pegasus test problem you're currently running into.

File: SETUP.md [hunk 2]
```diff
@@ -171,18 +171,30 @@ After setup, read about our [code styleguide](./STYLEGUIDE.md), our [test suites
...
 1. *(If working from an EC2 instance)* `sudo apt-get install -y libreadline-dev libffi-dev`
+1. configure your system so that `~/.bashrc` (or another startup file of your choice) will be run whenever you open a shell
+    1. if you are using bash and setting up a new linux system, follow [this explanation](https://joshstaiger.org/archives/2005/07/bash_profile_vs.html) and add the following lines to your `~/.bash_profile` config file:
+
+        ```
+        if [ -f ~/.bashrc ]; then
+          source ~/.bashrc
+        fi     
+        ```
+
```
Comment (Hamms) [inline SETUP.md]: note that newer systems use `.profile` rather than `.bash_profile`, and that often it's not necessary to set this up manually. What about something like 

```suggestion
1. configure your system so that `~/.bashrc` (or another startup file of your choice) will be run whenever you open a shell
    1. if you are using bash and setting up a new linux system, you may need to modify `~/.bash_profile` or `~/.profile` (your login shell configuration file) as per [this explanation](https://joshstaiger.org/archives/2005/07/bash_profile_vs.html), which recommends adding these lines:

        ```
        if [ -f ~/.bashrc ]; then
          source ~/.bashrc
        fi     
        ```
```

**Review by Hamms — COMMENTED** [review]
one nit, and I wanna second Tim's concern about pdftk. Otherwise LGTM

File: SETUP.md (hunk 1 above)
Comment (tim-dot-org) [reply SETUP.md:174]: I did have to update the dockerfile for pdftk, so maybe the instructions here should match what the dockerfile is doing, if the default apt-get isn't working
https://github.com/code-dot-org/code-dot-org/blob/fa1cdafb994273ba0044251c45d05659b5a10323/docker/dockerfiles/Dockerfile#L105

Comment (davidsbailey) [reply SETUP.md:174]: Thanks @tim-dot-org ! would you expect the same command to work when run from the command line on your system, without editing the docker file? 
```
curl -sSL -o /tmp/pdftk-java_3.0.9-1_all.deb https://mirrors.kernel.org/ubuntu/pool/universe/p/pdftk-java/pdftk-java_3.0.9-1_all.deb
```
if so, I can add it as an option here.

Comment (tim-dot-org) [reply SETUP.md:174]: yeah! it works on mine (Ubuntu 20) (and docker ubuntu 18)

Comment (davidsbailey) [reply SETUP.md:174]: whoa, I had no idea debian packages could be installed on ubuntu! this succeeds for me on Ubuntu 18.04 as well. unfortunately, it doesn't seem to eliminate the need for `pdftk`, and our pegasus tests still fail:
```
ubuntu@ip-10-0-0-21:~/src/cdo/pegasus$ rake test TEST=test/test_pdf_merger.rb

PDFMergerTest
  test_merge_local_from_file                                      FAIL (0.00s)
Minitest::Assertion:         Expected 'pdftk' to be installed.
            This test depends on pdftk (https://www.pdflabs.com/tools/pdftk-the-pdf-toolkit/)
            You should install it locally:
               OSX: brew install https://raw.github.com/quantiverge/homebrew-binary/pdftk/pdftk.rb
            Ubuntu: sudo apt-get install pdftk
        /home/ubuntu/src/cdo/pegasus/test/test_pdf_merger.rb:70:in `block in assert_local_dependencies'
        /home/ubuntu/src/cdo/pegasus/test/test_pdf_merger.rb:58:in `each'
        /home/ubuntu/src/cdo/pegasus/test/test_pdf_merger.rb:58:in `assert_local_dependencies'
        /home/ubuntu/src/cdo/pegasus/test/test_pdf_merger.rb:123:in `test_merge_local_from_file'
```
I'm not sure where you found the installation step for pdftk-java, but if you can find a similar one for `pdftk` I'll include it here. 

In the longer term, we should probably change our pegasus code to use pdftk-java instead since it sounds like it is replacing pdftk, after which point it'll be easier for us to install necessary dependencies locally. I'm not going to tackle that right now, though.

Comment (tim-dot-org) [reply SETUP.md:174]: oh! didn't notice it was the java version. that makes sense, the other workarounds you have should be good for now
